test/benchmarks/src/main/scala/scala/ArrayBenchmark.scala

```diff
@@ -0,0 +1,134 @@
+package scala
+
+import java.util.concurrent.TimeUnit
+
+import scala.util.Random
+
+import org.openjdk.jmh.annotations._
+
+@BenchmarkMode(Array(Mode.AverageTime))
+@Fork(2)
+@Threads(1)
+@Warmup(iterations = 10)
+@Measurement(iterations = 10)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@State(Scope.Benchmark)
+class ArrayBenchmark {
+
+  // From #1338: Optimize simple for loops
+
+  @Param(Array("10", "100", "1000"))
+  var loopSize: Int = _
+  var doubles: Array[Array[Double]] = _
+
+  @Setup(Level.Trial) def initArrays: Unit = {
+    // doubles = Array.ofDim[Double](loopSize,loopSize)
+    // doubles = Array.tabulate(loopSize, loopSize)((i,_) => i.toDouble)
+    doubles = Array.tabulate(loopSize, loopSize) { (_, _) =>
+      scala.math.sqrt(Random.nextDouble)
+    }
+  }
+
+  @Benchmark 
+  def matMulUsingIteratorsBench(): Unit = {
+    matMulUsingIterators(doubles, doubles, doubles)
+  }
+
+  def matMulUsingIterators(
```

Comment:
I find the name misleading, I don't see any Iterators here.

Reply:
Yes, I agree.  It's not the best name.